File: csharp/剑指06从尾到头打印链表.cs

public partial class Solution
{
    public class ListNode
    {
        public int val;
        public ListNode? next;

        public ListNode(int val)
        {
            this.val = val;
            next = null;
        }
    }
    public int[] ReversePrint(ListNode head)
    {
        int len = 0;
        for (ListNode p=head;p!=null ;p=p.next,len++);

        int[] res = new int[len];

        for (int i = len-1; i >= 0; --i,head=head.next)
        {
            res[i] = head.val;
        }

        return res;
    }
}

File: python/main/剑指06从尾到头打印链表.py
from linklist import ListNode


def reversePrint(head: ListNode) -> list:
    res=[]

    while(head!=None):
        res.append(head.val)
        head=head.next
    
    return res[::-1]



# 递归
def reversePrint2(head: ListNode) -> list:
    res=[]
    def _recursion(p:ListNode):    
        if(p==None):
            return
        _recursion(p.next)
        res.append(p.val)
        
    _recursion(head)
    return res 